client/src/models/populate_services.js

```diff
@@ -113,217 +223,137 @@ export function api_load_subject_has_services(subject) {
-          sin_collected
-          phone_inquiry_count
-          online_inquiry_count
-          online_application_count
-          live_application_count
-          mail_application_count
-          other_application_count
-          service_report_comment
-        }
+};
+const get_services_query = (query_options) => {
+  const { subject, query_fragments } = query_options;
+  const query_lookup_by_subject_level = {
+    gov: all_services_query,
+    dept: dept_services_query,
+    program: program_services_query,
+  };
+  return query_lookup_by_subject_level[subject.level](query_fragments);
+};
 
-        standards {
-          standard_id
-          service_id
-      
-          name
-      
-          last_gcss_tool_year
-          channel
-          type
-          other_type_comment
-      
-          target_type
-          urls
-          rtp_urls
-          standard_report {
-            standard_id
-            year
-            lower
-            count
-            met_count
-            is_target_met
-            standard_report_comment
-          }
+const get_query_appropirate_level = (subject) =>
+  subject.level === "dept" ? "org" : subject.level;
+
+const get_summary_query = (query_options) => {
+  const { subject, query_fragment } = query_options;
+  const query_by_level = {
+    gov: "gov",
+    dept: `org(org_id: "${subject.id}")`,
+    program: `program(id: "${subject.id}")`,
+  };
+  return gql`
+  query($lang: String!) {
+    root(lang: $lang) {
+      ${query_by_level[subject.level]} {
+        id
+        service_summary {
+          id
+          service_general_stats {
+            id
+            number_of_services
+          }  
+          ${query_fragment}
         }
       }
-  `;
-
-const dept_services_query = gql`
-query($lang: String!, $id: String) {
-  root(lang: $lang) {
-    org(org_id: $id) {
-      id
-      ${dept_service_fragment}
     }
   }
-}
-`;
+  `;
+};
 
-const all_services_query = gql`
-query($lang: String!) {
-  root(lang: $lang) {
-    orgs {
-      ${dept_service_fragment}
+export const useSingleService = (service_id) => {
+  const time_at_request = Date.now();
+  const query = gql`
+  query($lang: String!) {
+    root(lang: $lang) {
+      service(id: "${service_id}"){
+        ${all_service_fragments}
+      }
     }
   }
-}
-`;
-
-const extract_flat_data_from_hierarchical_response = (response) => {
-  const serviceStandards = [];
-  const services = _.chain(_.isArray(response) ? response : [response])
-    .map((resp) => resp.services)
-    .compact()
-    .flatten(true)
-    .each((service) =>
-      _.each(service.standards, (standard) =>
-        serviceStandards.push(_.omit(standard, "__typename"))
-      )
-    )
-    .value();
-  return { services, serviceStandards };
+  `;
+  const res = useQuery(query, {
+    variables: {
+      lang,
+    },
+  });
+  const { loading, error, data } = res;
+  if (error) {
+    const resp_time = Date.now() - time_at_request;
```

Comment:
Just pointing this out, I'm trying to find a good pattern for timing the `useQuery` hook on my own branch right now, but this isn't actually the time it took to fetch this data. When execution reaches this line, `time_at_request` is actually more like "time at response received" and `resp_time` is more like "execution time for lines 268 to 284".